### PR TITLE
repartition scenarios docs

### PR DESCRIPTION
**Pull request by simskij**
Repartition the scenarios docs into subpages to make it easier to digest and navigate. ~This PR intentionally does not include any significant changes to the content~. Edit: Scratch that, I'm touching the content as well! 😅

### CODE REVIEW

**Comment by simskij** [comment]
I think this is enough for now. @ppcano and @imiric, I'd love to get your feedback before I merge this.

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md [hunk 1]
```diff
@@ -55,726 +57,30 @@ export let options = {
 
 </div>
 
-### Common configuration
+## Common options
```
Comment (ppcano) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md:60]: `Common options` at this moment looks out of scope.

A possible solution could be to introduce earlier `Executors` and next `Common options`. 

For example, we point that each `executor` has its own configuration options but there are a set of common options shared between all the executor types, and describe the `Common` options section.

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/01 Executors.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+title: 'Executors'
+---
+
+> ### 🎉 New in v0.27.0
```
Comment (ppcano) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/01 Executors.md:5]: Feedback -> Besides adding this `note` to the `Scenarios` main page, I am not sure about adding it in all the other pages.

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/03 Graceful stop.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+of this property is `30s`.
+
+## Example
+
+<div class="code-group" data-props='{"labels": [ "graceful-stop.js" ], "lineNumbers": "[true]"}'>
+
+```js
+import http from 'k6/http';
+
+export let options = {
```
Comment (ppcano) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/03 Graceful stop.md:25]: I had to read twice to understand how it works. I think we can simplify the example.

1 -  Use `duration` https://k6.io/docs/using-k6/options#duration instead of  `stages`. 

2 - Can we use an executor that it does not need to use `gracefulRampDown`.

File: src/data/markdown/docs/01 guides/06 Misc/06 Glossary.md [hunk 4]
```diff
@@ -31,6 +29,12 @@ When discussing complex topics, it is usually a good idea to define a clear, sha
 
 An **HTTP Archive**, or **HAR file**, is a file containing logs of a browser interactions with the system under test. All of the included transactions are stored as JSON-formatted text. These archives may then be used to generate test scripts using, for instance, the [har-to-k6 Converter](https://github.com/loadimpact/har-to-k6). For more details, see the [HAR 1.2 Specification](http://www.softwareishard.com/blog/har-12-spec/).
 
+## Iteration
+
+An iteration is the same as an execution of the `default function`, or scenario `exec` function.
+
+You can either calculate iterations across all [virtual users](#virtual-users) (as done by the [Shared iterations](/using-k6/scenarios/executors/shared-iterations) executor), or per virtual user (as the [Per-VU Iterations](/using-k6/scenarios/executors/per-vu-iterations)).
```
Comment (ppcano) [inline src/data/markdown/docs/01 guides/06 Misc/06 Glossary.md:36]: Calculate or configure?

`You can either configure`

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/04 Arrival Rate.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+This is not ideal when the goal is to simulate a certain arrival rate of new VUs,
+or more generally throughput (eg. requests per second).
+
+## Open model
+
+To fix this problem we use an open model, decoupling the start of new VU iterations
+from the iteration duration and the influence of the target system's response time.
+
+![Arrival rate closed/open models](images/Scenarios/arrival-rate-open-closed-model.png)
```
Comment (ppcano) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/04 Arrival Rate.md:76]: This image is not loaded.

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/04 Arrival Rate.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+This tight coupling between the VU iteration duration and start of new VU iterations
+in effect means that the target system can influence the throughput of the test, via
+its response time. Slower response times means longer iterations and a lower arrival
+rate of new iterations, and vice versa for faster response times.
+
+In other words, when the target system is being stressed and starts to respond more
+slowly a closed model load test will play "nice" and wait, resulting in increased
+iteration durations and a tapering off of the arrival rate of new VU iterations.
+
```
Comment (ppcano) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/04 Arrival Rate.md:67]: Can we include a blockquote to make more clear the short summary.

>  In a Closed Model, the target system's overload can influence the number of iterations executed in your test.

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/04 Arrival Rate.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+```bash
+running (1m09.3s), 000/011 VUs, 60 complete and 0 interrupted iterations
+open_model ✓ [======================================] 011/011 VUs  1m0s  1 iters/s
+```
+
+Compared with the first example of the closed model, in this open model example we
+can see that the VU iteration arrival rate is now decoupled from the iteration duration.
+The response times of the target system are no longer influencing the load being
+put on the target system.
```
Comment (ppcano) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/04 Arrival Rate.md:121]: We can then use again a blockquote to summary the open model.

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/04 Arrival Rate.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+export let options = {
+  scenarios: {
+    open_model: {
+      executor: 'constant-arrival-rate',
+      rate: 1,
+      timeUnit: '1s',
+      duration: '1m',
+      preAllocatedVUs: 20,
+      maxVUs: 100,
```
Comment (ppcano) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/04 Arrival Rate.md:95]: I would remove `maxVUs` for simplification and because it is not necessary.

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/04 Arrival Rate.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+    },
+  },
+};
+
+export default function () {
+  // With the open model arrival rate executor config above,
+  // new VU iteration will start at a rate of 1 every second,
+  // and we can thus expect to get 60 iterations completed
+  // for the full 1m test duration.
+  http.get('http://httpbin.test.k6.io/delay/10');
```
Comment (ppcano) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/04 Arrival Rate.md:105]: I would use `http://httpbin.test.k6.io/delay/6` to show exactly the same function example as before.

**Review by ppcano — COMMENTED** [review]
-- Pepe's feedback

File: src/data/markdown/docs/01 guides/06 Misc/06 Glossary.md [hunk 10]
```diff
@@ -31,6 +29,12 @@ When discussing complex topics, it is usually a good idea to define a clear, sha
 
 An **HTTP Archive**, or **HAR file**, is a file containing logs of a browser interactions with the system under test. All of the included transactions are stored as JSON-formatted text. These archives may then be used to generate test scripts using, for instance, the [har-to-k6 Converter](https://github.com/loadimpact/har-to-k6). For more details, see the [HAR 1.2 Specification](http://www.softwareishard.com/blog/har-12-spec/).
 
+## Iteration
+
+An iteration is the same as an execution of the `default function`, or scenario `exec` function.
```
Comment (MStoykov) [inline src/data/markdown/docs/01 guides/06 Misc/06 Glossary.md:34]:
```suggestion
An iteration is an execution of the `default function`, or scenario `exec` function.
```

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md [hunk 11]
```diff
@@ -4,14 +4,16 @@ excerpt: ''
...
+> see the [k6 v0.27.0 release notes](https://github.com/loadimpact/k6/releases/tag/v0.27.0)).
 
-This is an optional feature and existing scripts and options will continue to work as before
-(with a few minor breaking changes, as mentioned
-in the [k6 v0.27.0 release notes](https://github.com/loadimpact/k6/releases/tag/v0.27.0)).
+## Benefits
 
-There are several benefits of using scenarios in your tests:
+Scenarios allow us to make in-depth configurations to how VUs and iterations are scheduled, allowing
+us to model diverse traffic patterns in our load tests. There are several benefits of using scenarios in your tests:
```
Comment (imiric) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md:16]: - "Scenarios allow us..." and "allowing us" sounds repetitive.
- Nit: the voice of the first sentence is 1st person plural, but the second one switches to 2nd person, which is a bit jarring. I understand we want to make the docs personable, but in many cases keeping an indirect voice is more approachable in technical documentation IMO, and pronouns are messy nowadays 😅 

I would rewrite this to:
```suggestion
k6 scenarios enable configuring how VUs and iterations are scheduled, making it possible to model diverse traffic patterns in load tests. There are several benefits of using scenarios:
```

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/01 Executors/04 ramping-vus.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+## Options
+
+In addition to the [common configuration options](/using-k6/scenarios#common-options) this executor
+also adds the following options:
+
+| Option             | Type    | Description                                                                   | Default |
+| ------------------ | ------- | ----------------------------------------------------------------------------- | ------- |
+| `startVUs`         | integer | Number of VUs to run at test start.                                           | `1`     |
+| `stages`           | array   | Array of objects that specify the target number of VUs to ramp up or down to. | `[]`    |
+| `gracefulRampDown` | string  | Time to wait for iterations to finish before starting new VUs.                | `"30s"` |
```
Comment (MStoykov) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/01 Executors/04 ramping-vus.md:26]:
```suggestion
| `gracefulRampDown` | string  | Time to wait for an already started iteration to finish before stopping it during a ramp down.                | `"30s"` |
```

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md (hunk 1 above)
Comment (imiric) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md:60]: This should still be an `h3` as it's part of the Configuration section.

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md [hunk 13]
```diff
@@ -4,14 +4,16 @@ excerpt: ''
 hideFromSidebar: false
 ---
 
-k6 v0.27.0 introduced the concept of scenarios: configurable modes of scheduling VUs
-and iterations, which can be used to model diverse traffic patterns in load tests.
+> ### 🎉 New in v0.27.0
+>
+> This feature is new as of version 0.27.0. Usage of this feature is optional and for the vast majority,
```
Comment (imiric) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md:9]:
```suggestion
> This feature is new as of version 0.27.0. Usage of this feature is optional and for the vast majority of users,
```

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md [hunk 14]
```diff
@@ -4,14 +4,16 @@ excerpt: ''
...
+> existing scripts and configurations will continue to work as before. For a list of breaking changes,
+> see the [k6 v0.27.0 release notes](https://github.com/loadimpact/k6/releases/tag/v0.27.0)).
```
Comment (imiric) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md:11]:
```suggestion
> see the [k6 v0.27.0 release notes](https://github.com/loadimpact/k6/releases/tag/v0.27.0).
```

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md [hunk 15]
```diff
@@ -4,14 +4,16 @@ excerpt: ''
...
 
-This is an optional feature and existing scripts and options will continue to work as before
-(with a few minor breaking changes, as mentioned
-in the [k6 v0.27.0 release notes](https://github.com/loadimpact/k6/releases/tag/v0.27.0)).
+## Benefits
```
Comment (imiric) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md:13]: I don't think this should be a header, since we haven't explained what scenarios are at this point. The paragraph below is both an introduction to the feature and a list of its benefits, so I think we're fine with just the preamble of "There are several benefits...".

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/01 Executors.md [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+| [Ramping VUs](/using-k6/scenarios/executors/ramping-vus)                     | `ramping-vus`           | A variable number of VUs execute as many<br/> iterations as possible for a specified amount of time.                                               |
+| [Constant Arrival Rate](/using-k6/scenarios/executors/constant-arrival-rate) | `constant-arrival-rate` | A fixed number of iterations are executed<br/> in a specified period of time.                                                                      |
+| [Ramping Arrival Rate](/using-k6/scenarios/executors/ramping-arrival-rate)   | `ramping-arrival-rate`  | A variable number of iterations are <br/> executed in a specified period of time.                                                                  |
+| [Externally Controlled](/using-k6/scenarios/executors/externally-controlled) | `externally-controlled` | Control and scale execution at runtime<br/> via [k6's REST API](/misc/k6-rest-api) or the [CLI](https://k6.io/blog/how-to-control-a-live-k6-test). |
```
Comment (imiric) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/01 Executors.md:25]: Why is this repeated here and on the Scenarios page? I would remove this page and leave the executor summary only on the Scenarios page.

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/01 Executors/01 shared-iterations.md [hunk 17]
```diff
@@ -0,0 +1,63 @@
+---
+title: 'Shared iterations'
+excerpt: ''
+---
+
+> ### 🎉 New in v0.27.0
+>
+> This feature is new as of version 0.27.0. Usage of this feature is optional and for the vast majority,
+> existing scripts and configurations will continue to work as before. For a list of breaking changes,
+> see the [k6 v0.27.0 release notes](https://github.com/loadimpact/k6/releases/tag/v0.27.0)).
```
Comment (imiric) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/01 Executors/01 shared-iterations.md:10]: Agree with Pepe that we shouldn't repeat this header everywhere and only show it on the main Scenarios page.

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/01 Executors/05 constant-arrival-rate.md [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+## Description
+
+A fixed number of iterations are executed in a specified period of time.
+Since iteration execution time can vary because of test logic or the
+system-under-test responding more slowly, this executor will try to compensate
+by running a variable number of VUs&mdash;including potentially initializing more in the middle
+of the test&mdash;in order to meet the configured iteration rate. This approach is
+useful for a more accurate representation of RPS, for example.
+
+See the [arrival rate](#arrival-rate) section for details.
```
Comment (imiric) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/01 Executors/05 constant-arrival-rate.md:21]:
```suggestion
See the [arrival rate](/using-k6/scenarios/arrival-rate) section for details.
```

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/01 Executors/07 externally-controlled.md [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+## When to use
+
+If you want to control the number of VUs while the test is running.
+
+Important: this is the only executor that is not supported in `k6 cloud`, it can only be used
+locally with `k6 run`.
+
+## Examples
+
+In this example, we'll execute a testrun controllable at runtime, starting with 0 VUs up to
```
Comment (imiric) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/01 Executors/07 externally-controlled.md:44]:
```suggestion
In this example, we'll execute a test controllable at runtime, starting with 0 VUs up to
```

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/03 Graceful stop.md [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+
+> ### 🎉 New in v0.27.0
+>
+> This feature is new as of version 0.27.0. Usage of this feature is optional and for the vast majority,
+> existing scripts and configurations will continue to work as before. For a list of breaking changes,
+> see the [k6 v0.27.0 release notes](https://github.com/loadimpact/k6/releases/tag/v0.27.0)).
+
+## Description
+
+This option is available for all executors except `externally-controlled` and allows the
```
Comment (imiric) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/03 Graceful stop.md:14]: I think we should add some background here about how k6 stopped iterations before v0.27.0, to explain why these options are useful.

Something like:

Prior to v0.27.0, k6 would interrupt any iterations in progress when the test duration was reached or when scaling down VUs with the `stages` option. In some cases this would lead to skewed metrics and unexpected test results. In v0.27.0 this behavior can be controlled with the `gracefulStop` and `gracefulRampDown` options.

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/04 Arrival Rate.md [hunk 21]
```diff
@@ -0,0 +1,121 @@
...
+## Open model
+
+To fix this problem we use an open model, decoupling the start of new VU iterations
+from the iteration duration and the influence of the target system's response time.
+
+![Arrival rate closed/open models](images/Scenarios/arrival-rate-open-closed-model.png)
+
+In k6 we've implemented this open model with our "arrival rate" executors. There are  
+two arrival rate executors to chose from for your scenario(s),
+[constant-arrival-rate](#constant-arrival-rate) and [ramping-arrival-rate](#ramping-arrival-rate):
```
Comment (imiric) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/04 Arrival Rate.md:80]: These should link to their respective pages.

File: src/data/markdown/docs/01 guides/02 Using k6/19 Javascript Compatibility Mode.md [hunk 22]
```diff
@@ -4,8 +4,6 @@ excerpt: ''
 hideFromSidebar: true
 ---
 
-> ### 🎉 This feature is new as of version 0.26.0
```
Comment (imiric) [inline src/data/markdown/docs/01 guides/02 Using k6/19 Javascript Compatibility Mode.md:7]: Do we want to delete these types of notes? I think they're useful to keep indefinitely as reference.

**Review by imiric — COMMENTED** [review]
LGTM overall, I like the split into several smaller pages. 👍

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md (hunk 11 above)
Comment (simskij) [reply src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md:16]: Makes total sense. Thanks!

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/01 Executors/05 constant-arrival-rate.md (hunk 18 above)
Comment (simskij) [reply src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/01 Executors/05 constant-arrival-rate.md:21]: Good catch. Thanks!

File: src/data/markdown/docs/01 guides/02 Using k6/19 Javascript Compatibility Mode.md (hunk 22 above)
Comment (simskij) [reply src/data/markdown/docs/01 guides/02 Using k6/19 Javascript Compatibility Mode.md:7]: Makes sense. However, I think it would make sense to incorporate them into the introduction in that case, as we'd otherwise get 🎉 's and alert boxes on every page given enough time 😅 .

File: src/data/markdown/docs/01 guides/06 Misc/06 Glossary.md (hunk 4 above)
Comment (simskij) [reply src/data/markdown/docs/01 guides/06 Misc/06 Glossary.md:36]: I think `calculate` makes more sense in this case as we're not talking about the configuration per se, but rather how they are measured/tallied. It's a fine balance between being too technical/going into implementation details, or keeping it high-level and conceptual.

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/04 Arrival Rate.md (hunk 8 above)
Comment (simskij) [inline src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/04 Arrival Rate.md:95]:
```suggestion
```

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md (hunk 1 above)
Comment (simskij) [reply src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md:60]: While you're correct, I think I'd like to keep it an h2 for now, as I want it to be displayed in the sidebar for smoother navigation. 🙇‍♂️

File: src/data/markdown/docs/01 guides/02 Using k6/19 Javascript Compatibility Mode.md (hunk 22 above)
Comment (simskij) [reply src/data/markdown/docs/01 guides/02 Using k6/19 Javascript Compatibility Mode.md:7]: Incorporated it into the text instead.

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/01 Executors.md (hunk 16 above)
Comment (simskij) [reply src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/01 Executors.md:25]: There's no good way to do this with our current setup in gatsby unfortunately. Each item in the sidebar requires it's own subpage or the menu generator goes bananas. u_u Will think of this is a future improvement.

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/04 Arrival Rate.md (hunk 7 above)
Comment (simskij) [reply src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/04 Arrival Rate.md:120]: Good suggestion! Rewrote it a bit and moved it to the top.

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/03 Graceful stop.md (hunk 3 above)
Comment (simskij) [reply src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios/03 Graceful stop.md:25]: Yes, fixed. 👍

File: src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md (hunk 1 above)
Comment (simskij) [reply src/data/markdown/docs/01 guides/02 Using k6/14 Scenarios.md:60]: Switched places of the sections. Good catch!
